src/tools/perf/ucp_tests.cc

```diff
@@ -196,43 +207,52 @@ class ucp_perf_test_runner {
         ucp_worker_h worker;
         ucp_ep_h ep;
         void *send_buffer, *recv_buffer;
+        ucp_datatype_t send_datatype, recv_datatype;
         uint64_t remote_addr;
         uint8_t sn;
         ucp_rkey_h rkey;
-        size_t length;
+        size_t length, send_length, recv_length;
 
-        length = ucx_perf_get_message_size(&m_perf.params);
+        length        = ucx_perf_get_message_size(&m_perf.params);
         ucs_assert(length >= sizeof(psn_t));
 
-        ucp_perf_test_prepare_iov_buffer();
+        ucp_perf_test_prepare_iov_buffers();
 
         *((volatile uint8_t*)m_perf.recv_buffer + length - 1) = -1;
 
         rte_call(&m_perf, barrier);
 
-        my_index = rte_call(&m_perf, group_index);
+        my_index      = rte_call(&m_perf, group_index);
 
         ucx_perf_test_start_clock(&m_perf);
 
-        send_buffer = m_perf.send_buffer;
-        recv_buffer = m_perf.recv_buffer;
-        worker      = m_perf.ucp.worker;
-        ep          = m_perf.ucp.peers[1 - my_index].ep;
-        remote_addr = m_perf.ucp.peers[1 - my_index].remote_addr + m_perf.offset;
-        rkey        = m_perf.ucp.peers[1 - my_index].rkey;
-        sn          = 0;
+        send_buffer   = m_perf.send_buffer;
+        recv_buffer   = m_perf.recv_buffer;
+        worker        = m_perf.ucp.worker;
+        ep            = m_perf.ucp.peers[1 - my_index].ep;
+        remote_addr   = m_perf.ucp.peers[1 - my_index].remote_addr + m_perf.offset;
+        rkey          = m_perf.ucp.peers[1 - my_index].rkey;
+        sn            = 0;
+        send_length   = length;
+        recv_length   = length;
+        send_datatype = ucp_perf_test_get_datatype(m_perf.params.ucp.send_datatype,
```

Comment:
@yosefe I didn't implement class variable for each datatype. I'm used it as local variables. I hope this variant is acceptable.
Also, I had to add `send_length` and `recv_length` to avoid influence on `length` with mixed datatype mode. Previously, when datatype things was in `send` and `recv` functions we has no such effect.